docs/design.md

```diff
@@ -0,0 +1,279 @@
+
+In order not to sync the whole chain of advertisements in every request and exclusively fetching the missing range, 
+the indexer node will have to be able to flag in the Graphsync request a "termination link",
+i.e. the link from which Graphsync can stop traversing the chain and stop recursing links.
+
+Once the sync process is finished, the indexer node updates the latest AdvertisementID seen for provider to `latest`.
+
+```go
+// Example of internal data structure of an index of data provider.
+type AdvertisementID Index_Link
+
+type Index struct {
+        Previous: Index_Link{},
+        Entries: []Entry{
+                IsRm:           bool, 
+                Cids:           []cid.Cid,        // NOTE: This can also be string.
+                Metadata:       []byte,
+        },
+} 
+```
+
+_NOTE: We need to build the selector or the IPLD storer in the indexer node so that it checks the latest Advertisement
+seen and it terminates Graphsync's DAG traversal conveniently._
+
+### IPLD-unaware Ingestion 
+> Out of scope of MVP. For the MVP we are only considering IPLD-aware ingestion, and we enforce the use of IPLD to providers.
+
+It implements a libp2p request-response ingestion protocol that indexer nodes can use when interacting with providers that are not indexing their
+data using IPLD, i.e. Advertisements are not directly linked to data. This protocol is inspired by 
+[Filecoin's Block Sync protocol](https://spec.filecoin.io/#section-algorithms.block_sync). The underlaying idea is the same, to "sync a chain of data".
+
+Thus, when calling `Sync(ctx, p, latest)`, indexer node sends a first request for ingestion as:
+```go
+req = IngestionRequest{
+        ID: latest,
+        Start: 0, 
+}
+```
+
+Provider will answer with the list of entries for `latest` pointing to the previous `AdvertisementID` in the `Advertisement` field. `Size` specifies
+the total number of entries that the advertisement include to give the indexer knowledge of when all entries for the update have been received.
+
+```go
+resp = IngestionResponse {
+        Size: 2
+        Error: 0
+        Start: 0
+        Advertisement: Advertisement{
+                ID: latest,
+                Previous: "latest-1",
+                IndexerID: Link_Index,
+                Provider: p,
+                Signature: []byte{},  // The original signature of the advertisement.
+                GraphSupport: false, 
+        }
+        Entries: {
+                []Entry{
+                        Cids: {Cid1, Cid2, ...},
+                        Metadata: []byte{},
+                }
+}
+```
+- If `Size > len(Entries)`, the indexer-node sends a new request to keep fetching data for `latest` with `Start: 1`
```

Comment:
The value of `Start` has to do with pagination, so that should be discussed in the next paragraph.  This paragraph is discussing chained responses.  See the [Chained Responses](https://www.notion.so/protocollabs/Indexer-Node-Design-4fb94471b6be4352b6849dc9b9527825#ba49552aa0bd47d192317a9f34e94555) section of the design doc for reference to make sure these match.

Reply:
That paragraph was supposed to describe how pagination +  chaining worked combined (but I think it failed :) ). I changed it to describe one after the other as it is in the design doc.